docs/aws/README.md

```diff
@@ -15,38 +44,34 @@ KARPENTER_VERSION=$(curl -fsSL \
   | jq -r '.tag_name')
 ```
 
-### Create a Cluster
+## Create a Cluster
 
-Karpenter can run anywhere, including on self-managed node groups, [managed node groups](https://docs.aws.amazon.com/eks/latest/userguide/managed-node-groups.html), or [AWS Fargate](https://aws.amazon.com/fargate/).
-This demo will run Karpenter on Fargate, which means all EC2 instances added to this cluster will be controlled by Karpenter.
+Create a cluster with `eksctl`. The below configuration file specifies a basic
+cluster (name, region), an IAM role for Karpenter to use, and two fargate
+profiles. 
 
-```bash
-curl -fsSL https://raw.githubusercontent.com/awslabs/karpenter/"${KARPENTER_VERSION}"/docs/aws/eks-config.yaml \
-  | envsubst \
-  | eksctl create cluster -f -
-```
+The two fargate profiles host `kube-system` and the Karpenter service itself.
+This permits Karpenter to manage all EC2 instances. 
 
-Tag the cluster subnets with the required tags for Karpenter auto discovery.
+Karpenter itself can run anywhere, including on self-managed node groups,
+[managed node groups](https://docs.aws.amazon.com/eks/latest/userguide/managed-node-groups.html), 
+or [AWS Fargate](https://aws.amazon.com/fargate/).
 
-> If you are using a cluster with version 1.18 or below you can skip this step.
-More [detailed here](https://github.com/awslabs/karpenter/issues/404#issuecomment-845283904).
+Karpenter will provision traditional instances on EC2. 
 
-```bash
-SUBNET_IDS=$(aws cloudformation describe-stacks \
-    --stack-name eksctl-${CLUSTER_NAME}-cluster \
-    --query 'Stacks[].Outputs[?OutputKey==`SubnetsPrivate`].OutputValue' \
-    --output text)
-
-aws ec2 create-tags \
-    --resources $(echo ${SUBNET_IDS//,/ }) \
-    --tags Key="kubernetes.io/cluster/${CLUSTER_NAME}",Value=
-```
+Additionally, the configuration file sets up an [OIDC
+provider](https://kubernetes.io/docs/reference/access-authn-authz/authentication/#openid-connect-tokens),
+necessary for IRSA (see below). Kubernetes supports OIDC as a standardized way
+of communicating with identity providers. 
 
-### Setup IRSA, Karpenter Controller Role, and Karpenter Node Role
-We recommend using [CloudFormation](https://aws.amazon.com/cloudformation/) and [IAM Roles for Service Accounts](https://docs.aws.amazon.com/eks/latest/userguide/iam-roles-for-service-accounts.html) (IRSA) to manage these permissions.
-For production use, please review and restrict these permissions for your use case.
+## Setup Authentication from Kubernetes to AWS (IRSA)
 
-> For IRSA to work your cluster needs an [OIDC provider](https://docs.aws.amazon.com/eks/latest/userguide/enable-iam-roles-for-service-accounts.html)
+IAM Roles for Service Accounts (IRSA) maps Kubernetes resources to roles
+(permission sets) on AWS. 
+
+First, define a role using the template below. It provides full access to EC2,
```

Comment:
I thought I left a comment here earlier about "full access to EC2". In case it got lost, I'll reiterate. We definitely don't do this, and it would be dangerous to do so.

Reply:
https://github.com/awslabs/karpenter/pull/506#discussion_r668975648

Reply:
can you help me understand the policy then? 

I'm only seeing this

```
          - Effect: Allow
            Principal:
              Service:
                !Sub "ec2.${AWS::URLSuffix}
```